libyul/optimiser/VarNameCleaner.h

```diff
@@ -0,0 +1,54 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include <libyul/AsmDataForward.h>
+#include <libyul/optimiser/ASTWalker.h>
+#include <unordered_map>
+#include <string>
+#include <boost/optional.hpp>
+
+namespace yul
+{
+
+class VarNameCleaner: public ASTModifier
+{
+public:
+	VarNameCleaner();
+
+	using ASTModifier::operator();
+	void operator()(VariableDeclaration& _varDecl) override;
+	void operator()(Identifier&) override;
+
+	static void run(Block& _ast);
+
+	/// @returns suffix-stripped name, if a suffix was detected, none otherwise.
+	static boost::optional<std::string> stripSuffix(std::string const& _name);
```

Comment:
Private?

Reply:
exactly. I made them public by intention (ran some tests), but forgot to make them private after that, again.